pkg/kubectl/cmd/util/config_flags.go

```diff
@@ -0,0 +1,363 @@
+	}
+	if f.Username != nil {
+		flags.StringVar(f.Username, flagUsername, *f.Username, "Username for basic authentication to the API server")
+	}
+	if f.Password != nil {
+		flags.StringVar(f.Password, flagPassword, *f.Password, "Password for basic authentication to the API server")
+	}
+	if f.ClusterName != nil {
+		flags.StringVar(f.ClusterName, flagClusterName, *f.ClusterName, "The name of the kubeconfig cluster to use")
+	}
+	if f.AuthInfoName != nil {
+		flags.StringVar(f.AuthInfoName, flagAuthInfoName, *f.AuthInfoName, "The name of the kubeconfig user to use")
+	}
+	if f.Namespace != nil {
+		flags.StringVarP(f.Namespace, flagNamespace, "n", *f.Namespace, "If present, the namespace scope for this CLI request")
+	}
+	if f.Context != nil {
+		flags.StringVar(f.Context, flagContext, *f.Context, "The name of the kubeconfig context to use")
+	}
+
+	if f.APIServer != nil {
+		flags.StringVarP(f.APIServer, flagAPIServer, "s", *f.APIServer, "The address and port of the Kubernetes API server")
+	}
+	if f.Insecure != nil {
+		flags.BoolVar(f.Insecure, flagInsecure, *f.Insecure, "If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure")
+	}
+	if f.CAFile != nil {
+		flags.StringVar(f.CAFile, flagCAFile, *f.CAFile, "Path to a cert file for the certificate authority")
+	}
+	if f.Timeout != nil {
+		flags.StringVar(f.Timeout, flagTimeout, *f.Timeout, "The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests.")
+	}
+
+	if f.RequireMatchedServerVersion != nil {
+		flags.BoolVar(f.RequireMatchedServerVersion, FlagMatchBinaryVersion, *f.RequireMatchedServerVersion, "Require server version to match client version")
+	}
+}
+
+func (f *ConfigFlags) WithDeprecatedPasswordFlag() *ConfigFlags {
+	f.Username = stringptr("")
+	f.Password = stringptr("")
+	return f
+}
+
+func NewConfigFlags() *ConfigFlags {
+	impersonateGroup := []string{}
+	insecure := false
+	requireMatchServerVersion := false
+
+	return &ConfigFlags{
+		Insecure:   &insecure,
+		Timeout:    stringptr("0"),
+		KubeConfig: stringptr(""),
+
+		ClusterName:                 stringptr(""),
+		AuthInfoName:                stringptr(""),
+		Context:                     stringptr(""),
+		Namespace:                   stringptr(""),
+		APIServer:                   stringptr(""),
+		CertFile:                    stringptr(""),
+		KeyFile:                     stringptr(""),
+		CAFile:                      stringptr(""),
+		BearerToken:                 stringptr(""),
+		Impersonate:                 stringptr(""),
+		RequireMatchedServerVersion: &requireMatchServerVersion,
+		ImpersonateGroup:            &impersonateGroup,
+	}
+}
+
+func stringptr(val string) *string {
+	return &val
+}
+
+type TestConfigFlags struct {
```

Comment:
This implementation looks very suspect.  I mean, why do they want an actual on disk discovery cache?  That can't be right.